spec/unit/widgets/base_spec.rb

```diff
@@ -0,0 +1,152 @@
+            @obj = Button.new(widget_controller, widget_filter)
+            aggregate_failures 'for .Button' do
+              expect(@obj).to be_instance_of(Button)
+              expect(@obj.instance_eval('@timeout', __FILE__, __LINE__)).to equal(5)
+              expect(@obj.instance_eval('@interval', __FILE__, __LINE__)).to equal(0.5)
+            end
+          end
+        end
+        context 'when be initialized with custom timeout' do
+          it 'has a class with new timeout' do
+            LibyuiClient.timeout = 2
+            @obj = Checkbox.new(widget_controller, widget_filter)
+            aggregate_failures 'for .Checkbox' do
+              expect(@obj).to be_instance_of(Checkbox)
+              expect(@obj.instance_eval('@timeout', __FILE__, __LINE__)).to equal(2)
+              expect(@obj.instance_eval('@interval', __FILE__, __LINE__)).to equal(0.5)
+            end
+          end
+        end
+        context 'when be initialized with custom interval' do
+          it 'has a class with new interval and still uses the new timeout' do
+            LibyuiClient.interval = 2
+            @obj = Combobox.new(widget_controller, widget_filter)
+            aggregate_failures 'for .Combobox' do
+              expect(@obj).to be_instance_of(Combobox)
+              expect(@obj.instance_eval('@timeout', __FILE__, __LINE__)).to equal(2)
+              expect(@obj.instance_eval('@interval', __FILE__, __LINE__)).to equal(2)
+            end
+          end
+        end
+      end
+      # end of Base
+
+      describe '#exists?' do
+        let(:response) { instance_double('Response', { body: '[{ "id": "foo", "debug_label": "Cancel" }]' }) }
+        subject { Base.new(widget_controller, widget_filter) }
+
+        context 'when filter exist' do
+          it 'returns true' do
+            expect(widget_controller).to receive(:find).and_return(response)
+            expect(subject.exists?).to be_truthy
+          end
+        end
+        context 'non-existent widget' do
+          it 'returns false' do
+            allow(widget_controller).to receive(:find) do
+              raise(Error::WidgetNotFoundError)
+            end
+            @obj = Base.new(widget_controller, { id: '' })
+            expect(subject.exists?).to be_falsy
+          end
+        end
+        # of exist?
+      end
+
+      describe '#property' do
+        let(:response) { instance_double('Response', { body: [{ id: 'foo', debug_label: 'Cancel' }] }) }
+        before do
+          allow(widget_controller).to receive(:find).and_return(response)
+          @obj = Base.new(widget_controller, widget_filter)
+        end
+        context 'when the property exists' do
+          it 'returns the value of the property' do
+            expect(@obj.property(:id)).to eq('foo')
+          end
+        end
+        # end of property
+      end
+
+      describe '#action' do
+        pending
+        #   let(:params) { { action: 'enter_text', value: 'changed' } }
```

Comment:
Let's not add any dead code.

Reply:
Done. Although that it left because it was a good example.

Reply:
Sure, then add it to the documentation in some section for writing tests.